rust/gui-client/src-tauri/src/client/gui/system_tray.rs

```diff
@@ -86,28 +85,22 @@ impl Tray {
         if Some(&menu) == self.last_menu_set.as_ref() {
             tracing::debug!("Skipping redundant menu update");
         } else {
-            self.app
-                .run_on_main_thread(move || {
-                    if let Err(error) = update(handle, &app, &menu) {
-                        tracing::error!(
-                            error = anyhow_dyn_err(&error),
-                            "Error while updating tray icon"
-                        );
-                    }
-                })
-                .context("Failed to update tray icon")?;
+            self.run_on_main_thread(move || {
+                firezone_logging::unwrap_or_debug!(
+                    update(handle, &app, &menu),
+                    "Error while updating tray menu: {}"
```

Comment:
Wild idea maybe we call this too quickly when resources change or something, leading to it crapping out. Leaving this comment here case this problem grows into a real issue.

Reply:
I have no idea to be honest. We only emit resources from connlib when they actually change so that shouldn't happen very often.